pkg/controllers/state/cluster.go

```diff
@@ -180,7 +181,32 @@ func (c *Cluster) newNode(node *v1.Node) *Node {
 		n.Capacity = n.InstanceType.Resources()
 	}
 	n.Available = resources.Subtract(c.getNodeAllocatable(node, n.Provisioner), resources.Merge(requested...))
-	return n
+}
+
+func (c *Cluster) populateVolumeLimits(node *v1.Node, n *Node) {
+	var csiNode storagev1.CSINode
+	if err := c.kubeClient.Get(c.ctx, client.ObjectKey{Name: node.Name}, &csiNode); err != nil && !errors.IsNotFound(err) {
+		logging.FromContext(c.ctx).Errorf("getting CSINode, %s", err)
```

Comment:
Since we're not follow normal error patterns, is it possible that we can get stuck due to a transient networking error?